### ISSUE TITLE
Ability to read offsets

### ISSUE THREAD

**Issue by mikeal** [ipfs/js-ipfs#348]
In order to do video playback properly I need to be able to seek through parts of a file.

So, what I basically need is a version of `ipfs.cat` that has offset support. Something like:

```javascript
ipfs.seek(hash, offset, length, (err, exporter) => {
  if (err) throw err
  exporter.pipe(destinationStream)
})
```

**Comment by mikeal** [comment]
It could also be called "range" if you want to match the http terminology.

**Comment by dignifiedquire** [comment]
I agree this would be really nice to have. There is some ground work to be done in the code, but it's very much possible to do (go-ipfs already supports this on the gateway, but not exposed through the api)

**Comment by diasdavid** [comment]
@dignifiedquire it is exposed on the Files API (`ipfs files read --offset`), but that requires moving the file to mfs, like:

```
> ipfs add file.txt
QmHash
> ipfs files cp /ipfs/QmHash /file.txt
> ipfs files read /file.txt --offset=200
```

This is related with the convergence of Files with the basic unixfs commands https://github.com/ipfs/specs/issues/98#issuecomment-236268043
